[WEBPAGE][READY-FOR-REVIEW]: Startup Department Page

williamntlam opened the pull request:
📖 **Description**

This PR features a design and implementation of a webpage for the Startup Department @ UTMIST.

This PR is related to Issue #28 

Figma Link: https://www.figma.com/design/ginzyRbJB0y9TW6yBxDYqa/Startup-Department-Page---UTMIST?node-id=0-1&p=f

✅ **Objectives**
- [x] _Layout & Wireframe (Frontend)_
   - [x] Create a responsive layout on Figma that includes:
      - [x] A Title "Innovation @ UTMIST"
      - [x] Description Text that discusses the purpose and ideas of Startup Department @ UTMIST
      - [x] An "Apply" button that redirects to the "Join Us" page
      - [x] A slider for Guest Speakers and Interested Investors: one each
   - [x] Styling & Theming:
      - [x] Follow the site’s existing design system (typography, color palette, spacing).
      - [x] Ensure each section is clearly separated (e.g., spacing or horizontal rule).
      - [x] Make headings bold and consistent in size:
- [ ] _Add Tests / QA Checklist_
   - [ ] Link to or reference our style-guide for any UI reviews.
   - [ ] Include a note in the issue that QA should verify each heading and placeholder 

🎯 **Acceptance Criteria**
- [ ] New page is live and follows all design rules 
- [ ] Appropriate tests are provided and all test cases are passing

**Additional Ideas/Notes**
- [ ] Have a downwards arrow to indicate to the user to keep scrolling (Could be useful later on)
- [ ] Make it mobile responsive
- [ ] The Apply button redirects the user to "/careers".
- [ ] Remember to remove the gradient background behind the images / slider.
- [ ] It might be good to make the content of the page smaller to match the width of the header.
- [ ] I can set up the code to make the changes required in the Fall easier - Things such as using a variable to make the 'Apply' button disappear and show an 'Applications closed' notice instead to make changes easier later on.

williamntlam committed:
feat: initialized startups page with URL /startups

williamntlam committed:
feat: implemented title and description of startup page in the hero section

williamntlam committed:
feat: implemented an apply here button that will redirect to the apply page along with a disclaimer of due date for applications [skip ci]

williamntlam committed:
style: added roboto font to parent container so that it applies to all text [skip ci]

williamntlam committed:
style: added an on hover colour transition for when the user hovers over the 'apply' button [skip ci]

williamntlam committed:
feat: implemented informational paragraph for the startups @ utmist part [skip ci]

williamntlam committed:
feat: hero section completed (ci/cd not skipped in this commit)

williamntlam committed:
fix: used '&#39;' instead of ' when used in text in the hero section

williamntlam committed:
style: outer corners of the image collage appears rounded to make it more visually appealing

williamntlam committed:
style: removed gradient behind pictures [skip ci]

williamntlam committed:
feat: added moving carousel for investors and guest speakers '

williamntlam committed:
style: centered text in moving carousel

williamntlam committed:
style: improved carousel

williamntlam committed:
style: changed format of webpage to display more vertically for vertical screens and horizontally for horizontal screens

williamntlam committed:
style: added more spacing between the slider components [skip ci]

williamntlam committed:
style: for the mobile view, if there is an odd number of investors or guess speakers the last card is cenetered [skip ci]

williamntlam committed:
style: made the overall page and text a bit smaller

williamntlam committed:
chore: added back navbar and header for ci/cd testing

williamntlam committed:
fix: replaced all <img> tags with <Image> and removed unused variables

williamntlam committed:
fix: made images in guess speaker / investor cards bigger

williamntlam committed:
style: centered text to the left of moving slider

williamntlam committed:
style: fixed some gap and padding errors

williamntlam commented:
Mobile Responsiveness Demo: https://youtu.be/ba1sfNvnP5I

williamntlam committed:
style: fixed slightly uncentered text for devices like ipads

ambroseling commented on client/src/app/layout.tsx:
I think you can remove these changes

williamntlam replied:
Just reverted the changes - my Prettier settings were changing single quotes to double quotes. Thanks for letting me know about this!

ambroseling commented:
hey William awesome work, just some minor comments on style

- I think it might be a good idea to break each image into its own container cuz each image looks a bit too small and its a bit hard to see the individual images, but you can use your creative freedom in terms of how you wanna organize the images
![Screenshot 2025-06-05 at 3 19 44 AM](https://github.com/user-attachments/assets/d652214c-0beb-4e63-9b10-af1b6f541a49)

- it might be a good idea to add some more horizontal margins here
![Screenshot 2025-06-05 at 3 19 19 AM](https://github.com/user-attachments/assets/ddc71f62-b2ef-46a2-9e18-b603751377f8)

- I think the hero section style should be kept consistent with the other pages but we should definitely have a reusable component for that across pages 
![Screenshot 2025-06-05 at 3 19 28 AM](https://github.com/user-attachments/assets/862a3573-4f20-431c-a120-7a7338d6c8a2)

- this may also need a bit more padding so that it doesn't overflow the border of the navbar
![Screenshot 2025-06-05 at 3 19 36 AM](https://github.com/user-attachments/assets/25c9bdb7-d793-4563-aef7-619f4e5f2f59)

I think we may need a later discussion on whether we want to include more info on this page i.e. features of our startup incubator program, why join us etc but that's outside of the scope of this PR, just an FYI

overall super awesome, tysm for the hard work :)

ambroseling reviewed (commented):
mostly looks good, see comments

williamntlam committed:
refactor: changed double quotations in layout.tsx to single quotations

williamntlam committed:
refactor: changed double quotations in layout.tsx to single quotations due to prettier settings

williamntlam committed:
refactor: removed redundant import of roboto font

williamntlam committed:
refactor: made a file called startups.ts in the types directory to contain types used in the startups page and moved the person interface to that file

williamntlam committed:
style: added horizontal padding to the investors + guess speaker cards

williamntlam committed:
style: added padding between the hero component and the navbar

williamntlam committed:
style: increased sizes of images

williamntlam committed:
style: made container edges containing images rounded

williamntlam committed:
chore: added back navbar and footer

williamntlam committed:
style: made images the same size

williamntlam commented:
Hi @ambroseling,

Thanks for the comments, I definitely overlooked some styling stuff so I appreciate it. 

I've made the following changes:
- Put each image into its own container and made them slightly bigger.
- Added horizontal padding to the investor and guess speaker cards.
- Added padding to the hero section so that it doesn't overflow over to the navbar. 

I hope that helps,

William.

KCui0327 reviewed (approved):
I tested on my end, it looks fantastic! Good changes to feedback, it LGTM

williamntlam committed:
Merge branch 'main' into webpage/28-startup-department-page

ambroseling commented:
<img width="573" alt="Screenshot 2025-06-07 at 9 12 11 PM" src="https://github.com/user-attachments/assets/d2fca01a-875c-4abd-953c-3a1fd8d8c870" />
everything looks good only thing is perhaps adding a bit more padding on the left and right side of the screens

williamntlam committed:
Merge branch 'main' into webpage/28-startup-department-page

williamntlam committed:
style: added more padding on both sides of the page for medium screens
